openml/datasets/functions.py

```diff
@@ -411,10 +447,25 @@ def create_dataset(name, description, creator, contributor, collection_date,
     -------
     class:`openml.OpenMLDataset`
         Dataset description."""
+
+    if attributes == 'auto':
+        if not hasattr(data, "columns"):
+            raise ValueError("Automatically inferring the attributes required "
+                             "a pandas DataFrame. A {!r} was given instead."
+                             .format(data))
+        # infer the type of data for each column of the DataFrame
+        attributes_ = [(col_name,
+                        _pandas_dtype_to_arff_dtype(data, col_name, col_dtype))
+                       for col_name, col_dtype in data.dtypes.iteritems()]
```

Comment:
you don't necessarily need to pass the dtype here, you could also iterate only through the column names